test/test_trie.jl

```diff
@@ -1,7 +1,7 @@
-using DataStructures
+include("../src/trie.jl")
```

Comment:
The include should happen in src/DataStructures.jl :)

There's also some tests and stuff still in the suffixtrie file.

SuffixTrie is a neat datastructure, would be great to have!